cmd/gapic-showcase/compliance_suite_negative_test.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"testing"
+
+	"github.com/googleapis/gapic-showcase/server/genproto"
+	pb "github.com/googleapis/gapic-showcase/server/genproto"
+	"google.golang.org/protobuf/encoding/protojson"
+)
+
+// TestComplianceSuiteErrors checks for non-spec-compliant HTTP requests. Not all of these
+// conditions necessarily generate a server error in a real service, but the behavior is often
+// ill-defined. We want Showcase to require the generators be strict in the transcoding format they
+// use.
+func TestComplianceSuiteErrors(t *testing.T) {
+	// return
+	// Run the Showcase REST server locally.
+	server := httptest.NewUnstartedServer(nil)
+	backend := createBackends()
+	restServer := newEndpointREST(nil, backend)
+	server.Config = restServer.server
+	server.Start()
+	defer server.Close()
+
+	// Locate, load, and unmarshal the compliance suite.
+	_, thisFile, _, _ := runtime.Caller(0)
+	suiteFile := filepath.Join(filepath.Dir(thisFile), "../../schema/google/showcase/v1beta1/compliance_suite.json")
+	jsonProto, err := ioutil.ReadFile(suiteFile)
+	if err != nil {
+		t.Fatalf("could not open suite file %q", suiteFile)
+	}
+	var suite pb.ComplianceSuite
+
+	if err := protojson.Unmarshal(jsonProto, &suite); err != nil {
+		t.Fatalf("error unmarshalling from json %s:\n   file: %s\n   input was: %s", err, suiteFile, jsonProto)
+	}
```

Comment:
Can this stuff be moved into a function or two? Both this test file and the other use the same code (roughly the same I guess I haven't gone line-by-line). Maybe one to create a test server & return it (to then be started & the close deferred) and another for the compliance suite proto loading? Or one `setup` function that returns a server and the loaded suite?